test/unit/BPool/BPool_Bind.tree

```diff
@@ -0,0 +1,27 @@
+BPool_Bind::whenReentrancyLockIsNOTSet
+├── when caller is NOT controller
+│   └── it should revert
+└── when caller is controller
+    ├── when token is already bound
+    │   └── it should revert
+    ├── when pool is finalized
+    │   └── it should revert
+    ├── when MAX_BOUND_TOKENS tokens are already bound
+    │   └── it should revert
+    ├── when token weight is too low
+    │   └── it should revert
+    ├── when token weight is too high
+    │   └── it should revert
+    ├── when too little balance is provided
+    │   └── it should revert
+    ├── when weight sum exceeds MAX_TOTAL_WEIGHT
+    │   └── it should revert
+    └── when token can be bound
+        ├── it sets reentrancy lock
+        ├── it increments _totalWeight
+        ├── it calls _pullUnderlying
+        ├── it adds token to the list
+        └── it sets the token record
+
+BPool_Bind::whenReentrancyLockIsSet
+└── it should revert
```

Comment:
hmmm polemic, why not adding just a leaf at the top as `when caller is NOT controller` but saying `when reentrancy is locked > it should revert`?